modules/liblc3/CMakeLists.txt

```diff
@@ -1,22 +1,32 @@
 if(CONFIG_LIBLC3)
-
 zephyr_library_named(liblc3)
-zephyr_library_compile_options(-O3 -std=c11 -ffast-math -Wno-array-bounds)
+zephyr_library_compile_options(-O3 -std=c11 -Wall -ffast-math -Wextra -Wdouble-promotion -Wvla -pedantic)
+
+zephyr_library_compile_options_ifndef(CONFIG_LIBLC3_PLUS
+	-DLC3_PLUS=0
+)
 
-zephyr_include_directories(${ZEPHYR_LIBLC3_MODULE_DIR}/include)
-zephyr_include_directories(${ZEPHYR_LIBLC3_MODULE_DIR}/src)
+zephyr_library_compile_options_ifndef(CONFIG_LIBLC3_PLUS_HR
+	-DLC3_PLUS_HR=0
+)
```

Comment:
Isn't this double negative? 

Instead of saying `if CONFIG_LIBLC3_PLUS_HR then -DLC3_PLUS_HR=1` you are saying `if not CONFIG_LIBLC3_PLUS_HR then -DLC3_PLUS_HR=0`.

Why not the former approach?

Reply:
`LC3_PLUS_HR` and `LC3_PLUS` are enabled by default. https://github.com/google/liblc3/blob/bb85f7dde4195bfc0fca9e9c7c2eed0f8694203c/src/common.h#L38
In this PR I reverse the logic, so that it needs to be explicitly enabled by the user.

Reply:
Ah.

I suggest that we add that as a comment here in this file, so that it's clear. 
We can also set them to `0` by default, and then to 1 when the respective Kconfig options are set.